source/objects/lander.lua

```diff
@@ -30,9 +31,67 @@ local engineSound = Assets.getSound("engine")
 -- Local functions
 -- ~~~~~~~~~~~~~~~~
 
+
+
+local function recalcDefaultMass(lander)
+	local result = 0
+	-- all the masses are stored in this table so add them up
+	for i = 1, #lander.mass do
+		result = result + lander.mass[i]
+	end
+	-- return mass of all the components + mass of fuel
+	return (result + lander.fuelCapacity)
+end
+
+
+
+local function landerHasFuelToThrust(lander, dt)
+-- returns true if the lander has enough fuel for thrust
+-- returns false if not enough fuel to thrust
+-- Note: fuel can be > 0 but still not enough to thrust
+
+	local hasThrusterUpgrade = Lander.hasUpgrade(lander, Modules.thrusters)
+	if (lander.fuel - dt) >= 0 or (hasThrusterUpgrade and (lander.fuel - (dt * 0.80)) >= 0) then
+		return true
+	else
+		return false
+	end
+end
+
+
+
+local function parachuteIsDeployed(lander)
+-- return true if lander has a parachute and it is deployed
+
+	for _, moduleItem in pairs(lander.modules) do
+		if moduleItem.id == 5 then	-- 5 = parachute	
+			if moduleItem.deployed then
+				return true
+			end
+		end
+	end
+	return false
+end
```

Comment:
If we change the `hasUpgrade` function to `getUpgrade` (which should be the same with the difference that the return value will be nil or a module) then we can reduce this to
```lua
function Lander.getUpgrade(lander, module)
	for i = 1, #lander.modules do
		if lander.modules[i] == module then
			return lander.modules[i]
		end
	end
end


local function parachuteIsDeployed(lander)
	local module = Lander.getUpgrade(lander, Modules.parachute)
	if module then
		return module.deployed
	end
end
```
Not the cleanest solution either way but I think that would be better.